internal/discourse/discourse.go

```diff
@@ -0,0 +1,88 @@
+package discourse
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+)
+
+const (
+	internalTestCategory  = 29
+	buildIssuesCat = 11
+)
+
+var (
+	discourseAPIKey     = os.Getenv("DISCOURSE_API_KEY")
+	discourseAPIUser    = os.Getenv("DISCOURSE_API_USER")
+	discourseCategoryID int
+	topicTpl = `Original GitHub post: %s
+	
+	%s`
+)
+
+func init() {
+	if discourseAPIKey == "" {
+		fmt.Println("DISCOURSE_API_KEY empty")
+		os.Exit(1)
+	}
+
+	if discourseAPIUser == "" {
+		fmt.Println("DISCOURSE_API_USER empty")
+		os.Exit(1)
+	}
+
+	flag.IntVar(&discourseCategoryID, "discourse-category-id", internalTestCategory, "--discourse-category-id=<int> (discourse category to post topics to)")
+}
+
+func PostTopic(title string, originURL, content string) (string, error) {
+	message := make(map[string]interface{})
```

Comment:
Note: if you are not using make but creating the variable in-place then you can have the following lines included in the var creating as well as initial values.

Reply:
for example:
```
message := make(map[string]interface{
	"title": title,
	"category": discourseCategoryID,
	"raw": fmt.Sprintf(topicTpl, originURL, content),
})
```